cursor.go

```diff
@@ -29,11 +29,19 @@ func (c *Cursor) Bucket() *Bucket {
 // If the bucket is empty then a nil key and value are returned.
 // The returned key and value are only valid for the life of the transaction.
 func (c *Cursor) First() (key []byte, value []byte) {
+	k, v, flags := c.first()
+	if (flags & uint32(bucketLeafFlag)) != 0 {
+		return k, nil
+	}
+	return k, v
+}
+
+func (c *Cursor) first() (key []byte, value []byte, flags uint32) {
 	_assert(c.bucket.tx.db != nil, "tx closed")
```

Comment:
Suggest to keep the existing check behavior: only check the `_assert(c.bucket.tx.db != nil, "tx closed")` on the exported Methods (e.g. First, Last, Next, Prev).

But [seek](https://github.com/etcd-io/bbolt/blob/dd0ab6da682ad81a885aeca8bb1fd3b6a9ff8d54/cursor.go#L147) is an abnormal example. We may need to move the check to `Seek`.

Reply:
https://github.com/etcd-io/bbolt/pull/363

Reply:
Done.